File: 2458-find-minimum-time-to-finish-all-jobs-ii/2458-find-minimum-time-to-finish-all-jobs-ii.py
class Solution:
    def minimumTime(self, jobs: List[int], workers: List[int]) -> int:
        # Sort both the arrays
        jobs.sort()
        workers.sort()
        ans = 0

        # Find the length to iterate through both
        job_lenth = len(jobs)

        for i in range(job_lenth):
            days = jobs[i]
            capacity = workers[i]

            # If workers are more or equal to jobs
            if capacity >= days:
                ans = max(ans, 1)
            else:
                # If workers are less than jobs so get multiplier
                total_days = (days // capacity) if (days % capacity) == 0 else (days // capacity) + 1
                ans = max(ans, total_days)
        return ans

File: 2458-find-minimum-time-to-finish-all-jobs-ii/2458-find-minimum-time-to-finish-all-jobs-ii.java
class Solution {
    public int minimumTime(int[] jobs, int[] workers) {

        // Sorth both the array in ascending order
        Arrays.sort(jobs);
        Arrays.sort(workers);

        int maxJobDays = 0;
        int currentJobDays = 0;

        // Iterate through both the arrays
        for(int i =0 ; i<jobs.length; i++){
            // This helps to get the days required for workers even in workers are greater than jobs
            // Or even if its a multiplication factor of jobs
            currentJobDays = (jobs[i]+workers[i]-1)/workers[i];
            maxJobDays = Math.max(maxJobDays, currentJobDays);
        }

        return maxJobDays;
        
    }
}